### PR TITLE
DrivAerML: metric-aware w=0.04 + WD=1e-3 (weight+regularization)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
Combine intermediate metric-aware weight (w=0.04) with weight decay (1e-3). Mugen #3328 tests w=0.05+WD; this tests a lower weight to see if WD allows lighter rel-L2 touch. May close the val-test gap.

## Instructions
```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml --dm-loss mse_plus_raw_rel_l2 --dm-rel-l2-weight 0.04 \
  --optimizer adamw --lr 5e-4 --weight-decay 1e-3 --cosine-t-max 30 --grad-clip 0.5 \
  --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 \
  --ema-decay 0.9995 --ema-mode fixed --no-compile-model --save-checkpoint \
  --eval-interval 10 --seed 42 --wandb-group dm-metric-aware-sweep \
  --kill-if-best-val-above 150:8.0 --kill-if-no-improvement 200:0.05
```
**Target:** val < 3.700% or test < 4.320%

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT fern:

## Results

### DrivAerML: metric-aware w=0.04 + WD=1e-3

| Metric | This Run | Baseline (PR #3302, w=0.05, no WD) | Delta |
|--------|----------|--------------------------------------|-------|
| **val_primary/surface_rel_l2_pct** | **3.874%** (ep 900) | **3.700%** (ep 780) | +0.174% ❌ |
| **test_primary/surface_rel_l2_pct** (best-ckpt) | **4.549%** | **4.500%** (batch-limited) | +0.049% ❌ |
| **test_primary/surface_rel_l2_pct** (final ep 999) | **4.647%** | — | — |
| External target (AB-UPT) | — | <3.71% | — |

### Config
- `--dm-loss mse_plus_raw_rel_l2 --dm-rel-l2-weight 0.04`
- `--optimizer adamw --lr 5e-4 --weight-decay 1e-3 --cosine-t-max 30 --grad-clip 0.5`
- `--model-layers 4 --model-hidden-dim 512 --model-heads 8 --enable-fourier`
- `--ema-decay 0.9995 --ema-mode fixed --no-compile-model --seed 42`
- `--batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200`

### Command
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml --dm-loss mse_plus_raw_rel_l2 --dm-rel-l2-weight 0.04 \
  --optimizer adamw --lr 5e-4 --weight-decay 1e-3 --cosine-t-max 30 --grad-clip 0.5 \
  --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 \
  --ema-decay 0.9995 --ema-mode fixed --no-compile-model --save-checkpoint \
  --eval-interval 10 --seed 42 --wandb-group dm-metric-aware-sweep \
  --kill-if-best-val-above 150:8.0 --kill-if-no-improvement 200:0.05 \
  --agent fern --wandb-name "fern/dm-w004-wd1e3"
```

### W&B
- **Run ID:** `rus8uqqm`
- **URL:** https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/rus8uqqm

### Peak Memory
~7.4 GB GPU VRAM (GPU 0)

### What happened
**The hypothesis did not hold.** Lowering the rel-L2 weight from 0.05 to 0.04 while adding weight decay (1e-3) did not improve over the baseline. Both val and test metrics are worse:

- **Val:** 3.874% vs 3.700% baseline (+0.174%) — the lighter rel-L2 touch (w=0.04) appears to provide less metric alignment than w=0.05, and WD did not compensate.
- **Test:** 4.549% vs 4.500% baseline (+0.049%) — marginal degradation.
- The model trained for 999 epochs (vs baseline's 780 at best), hitting the epoch cap. Despite more training time, it never matched the baseline's val performance.
- Convergence was stable throughout with no divergence or instability. The model plateau'd around 3.87-3.97% from ~epoch 700 onwards, oscillating within that band.

**Weight decay with a lower w appears to be net negative.** The regularization from WD may have constrained the model's ability to fit the metric-aware loss component. The baseline's w=0.05 without WD allows the model to specialize more aggressively on the rel-L2 metric.

### Suggested follow-ups
- **w=0.05 + WD=1e-3** (Mugen #3328 may already be testing this) — WD at the baseline weight may behave differently.
- **w=0.04 without WD** — isolate whether the performance drop is from lower weight or WD.
- **WD=5e-4 or 1e-4 with w=0.05** — lighter WD may provide regularization benefits without constraining metric-aware loss fitting.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3334 because ADVISOR: Closing — val=3.874% with w=0.04+WD at T_max=30. Below T_max=36 baseline (3.622%). All T_max=30 metric-aware experiments are now superseded..
